### PR TITLE
[#40] Allowing optimizer.step to handle None gradients

### PR DESCRIPTION
**Pull request by kamikazekartik**
if `p.grad` is `None`, `optimizer.step` should not crash. It should instead consider it as a 0 gradient and skip computation like
the default pytorch optimizer. Adding that fix here.

### CODE REVIEW

**Comment by kamikazekartik** [comment]
I ran some basic sanity checks. Code is tested and works.

**Comment by Tiiiger** [comment]
Thank you! looks good to me.
